doc: sphinx-lint: fix bad usage of "default role" (single backticks)

kartben opened the pull request:
This PR removes the incorrect use of single backticks across all documentation files. We haven't had support for Sphinx's "any" default role for a long time (PR #41061) so any remaining single backticks were either due to people wanting "inline literal" formatting (i.e. **double** backticks), or *italics* (single `*`).

When appropriate, a better construct than double backticks has been selected (ex. `:file:`, `:kconfig:option:`, `:c:func:`, ...) or proper `:ref:` has been used if the original intention was to have a link.

The actual fix is in third commit.
The first two commits contain other minor lint fixes.
Last commit drops the section in the documentation guidelines that was still referring to the default any role / single backticks. 

**Next step after this is merged will be to introduce a Sphinx linter (#78292), as there will basically be no lint errors left anymore 🥳**

kartben committed:
doc: sphinx-lint: fix bad usage of "default role"

Fixes bad usage of single backticks in lieu of double backticks for
rendering inline literals, or simple '*' for italics.

When appropriate, a better construct than double backticks has been
selected (ex. :file:, :kconfig:option:, :c:func:, ...), or proper :ref:
have been used if the original intention was to have a link.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: guidelines: drop mention of "any" links

Commit 204860857edbcfd4a57928acae8a64ee0e43f00a (PR #41061) dropped
support for "any" as the default role, so this should be dropped from
the documentation guidelines as well.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

Thalley commented on boards/sparkfun/micromod/doc/index.rst:
```diff
@@ -139,7 +139,7 @@ applications as usual (see :ref:`build_an_application` and
+In the case of `Sparkfun asset tracking carrier`_, it is possible to use
 the SWD interface along with a J-Link.
 
 Here is an example for the :ref:`hello_world` application.
```
Ditto with sample ref?

kartben replied:
see above

Thalley commented on boards/sparkfun/thing_plus/doc/index.rst:
```diff
@@ -92,7 +92,7 @@ In most cases you'll want to use the ``ns`` target with any of the Zephyr
 
 Some of the examples do not use secure mode, so they do not required the ``ns`` suffix.
-A great example of this is the `hello_world` below.
+A great example of this is the :ref:`hello_world` below.
```
Ditto (won't bring this more times even if I spot it more)

Thalley commented on boards/st/b_u585i_iot02a/doc/index.rst:
```diff
@@ -236,7 +236,7 @@ option bit TZEN will be set).
 
 Please note that, after having run a TFM sample on the board, you will need to
-run `./build/tfm/api_ns/regression.sh` once more to clean up the board from secure
+run ``./build/tfm/api_ns/regression.sh`` once more to clean up the board from secure
```
File ref?

kartben replied:
or command? 😃

Thalley commented on boards/st/nucleo_l552ze_q/doc/nucleol552ze_q.rst:
```diff
@@ -197,9 +197,9 @@ The BOARD options are summarized below:
 
 Here are the instructions to build Zephyr with a non-secure configuration,
-using `tfm_ipc_` sample:
+using :ref:`tfm_ipc` sample:
```
sample ref?

Thalley commented on boards/st/nucleo_l552ze_q/doc/nucleol552ze_q.rst:
```diff
@@ -213,7 +213,7 @@ option bit TZEN will be set).
 
 Please note that, after having run a TFM sample on the board, you will need to
-run `./build/tfm/api_ns/regression.sh` once more to clean up the board from secure
+run ``./build/tfm/api_ns/regression.sh`` once more to clean up the board from secure
```
File ref?

Thalley commented on boards/st/stm32l562e_dk/doc/index.rst:
```diff
@@ -223,7 +223,7 @@ The BOARD options are summarized below:
 
 Here are the instructions to build Zephyr with a non-secure configuration,
-using `tfm_ipc_` sample:
+using :ref:`tfm_ipc` sample:
```
Ditto

Thalley commented on boards/st/stm32l562e_dk/doc/index.rst:
```diff
@@ -239,7 +239,7 @@ option bit TZEN will be set).
 
 Please note that, after having run a TFM sample on the board, you will need to
-run `./build/tfm/api_ns/regression.sh` once more to clean up the board from secure
+run ``./build/tfm/api_ns/regression.sh`` once more to clean up the board from secure
```
Ditto

Thalley commented on boards/st/stm32wb5mmg/doc/stm32wb5mmg.rst:
```diff
@@ -248,8 +248,8 @@ the ``--runner`` (or ``-r``) option:
 
-Flashing `hci_uart` application to STM32WB5MMG
-----------------------------------------------
+Flashing ``hci_uart`` application to STM32WB5MMG
```
Sample ref?

Thalley commented on doc/develop/manifest/index.rst:
```diff
@@ -12,7 +12,7 @@ Active Projects/Modules
 
 The projects below are enabled by default and will be downloaded when you
-call `west update`. Many of the projects or modules listed below are
+call :command:`west update`. Many of the projects or modules listed below are
```
Curiously here we use `:command:` for west :D

Thalley commented on doc/out.webp:
What happened in this file?

kartben replied:
thanks for spotting this!

Thalley commented on doc/services/device_mgmt/ec_host_cmd.rst:
```diff
@@ -120,10 +120,10 @@ Logging
-* `LOG_INF` is used to log a command id of a new command and not success responses. Repeats of the
+* ``LOG_INF`` is used to log a command id of a new command and not success responses. Repeats of the
   same command are not logged
-* `LOG_DBG` logs every command, even repeats
-* `LOG_DBG` + :kconfig:option:`CONFIG_EC_HOST_CMD_LOG_DBG_BUFFERS` logs every command and responses
+* ``LOG_DBG`` logs every command, even repeats
+* ``LOG_DBG`` + :kconfig:option:`CONFIG_EC_HOST_CMD_LOG_DBG_BUFFERS` logs every command and responses
```
There are functions/macros; should they use the macro or func refs?

kartben replied:
absolutely -- an oversight :)

Thalley commented on samples/bluetooth/peripheral_hids/README.rst:
```diff
@@ -15,7 +15,7 @@ In the default configuration the sample uses passkey authentication (displays a
 pairing) and requires an authenticated link to access the GATT characteristics.
 To disable authentication and just use encrypted channels instead, build the
-sample with `CONFIG_SAMPLE_BT_USE_AUTHENTICATION=n`.
+sample with :kconfig:option:`CONFIG_SAMPLE_BT_USE_AUTHENTICATION` ``=n``.
```
Does this render OK?

kartben replied:
probably not ideal (see ex. [CONFIG_SAMPLE_BT_USE_AUTHENTICATION](https://docs.zephyrproject.org/latest/connectivity/bluetooth/bluetooth-arch.html#build-types) , no, but I messed up anyway: CONFIG_SAMPLE_BT_USE_AUTHENTICATION is local to this sample so there is no point in using the Sphinx role. I will just convert the single backticks to double backticks

Thalley replied:
So `:kconfig:option:` should not be used for sample defined Kconfig values?

kartben replied:
It *can* be used, it just doesn't really make a difference vs. using double backticks. In theory I guess we could check if those "local" Kconfig options exist, but that sounds overkill.

Thalley commented on samples/drivers/fpga/fpga_controller/README.rst:
```diff
@@ -82,8 +82,8 @@ To upload the bitstream again you need to reset the FPGA:
 You can also use your own bitstream.
-To load a bitstream into device memory, use `devmem load` command.
-It is important to use the -e option when sending a bitstream via `xxd`:
+To load a bitstream into device memory, use ``devmem load`` command.
```
Ditto with command

Thalley commented on samples/sysbuild/with_mcuboot/README.rst:
```diff
@@ -19,7 +19,7 @@ sysbuild.
 :file:`sysbuild.conf`.
 
-The `SB_CONFIG_BOOTLOADER_MCUBOOT=y` setting in the sysbuild Kconfig file
+The ``SB_CONFIG_BOOTLOADER_MCUBOOT=y`` setting in the sysbuild Kconfig file
```
Is this not considered a Kconfig option?

kartben replied:
I have to admit I have no idea how sysbuild kconfigs work. The idea behind `:kconfig:option:` is that it creates a cross-reference to the Kconfig listing. But in this case SB_CONFIG_BOOTLOADER_MCUBOOT is *not* listed there: https://docs.zephyrproject.org/latest/kconfig.html#SB_CONFIG_BOOTLOADER_MCUBOOT
We could still use the role to convey the idea that it is a kconfig option (just like we could also systematically use the role even for those local `CONFIG_SAMPLE_XYZ` option), it would just not link to anything (but still render as a literal)

kartben commented on boards/raspberrypi/rpi_5/doc/index.rst:
```diff
@@ -125,14 +126,14 @@ config.txt
 ----------
 
-Build an app `samples/hello_world`
+Build an app, for example :ref:`hello_world`:
```
Believe it or not, "hello world" hasn't been converted yet :) 
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/hello_world/README.rst?plain=1#L1-L4

de-nordic reviewed (dismissed):
MCUmgr doc looks ok.

kartben committed:
doc: releases: fix kconfig role typo in mbedtls section

trivial typo fix where single backticks should be used instead of double
for the sphinx role kconfig:option

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben commented:
Addressed merge conflict in `samples/sample_definition_and_criteria.rst` and fixed a few new recent lint errors in release notes and migration guide.

kartben commented:
Had to rebase due to a new annoying merge conflict were I literally had nothing to resolve when rebasing. It just looks like Github was having a hard time with a recent keeping track of files subject to a recent `git mv` (https://github.com/zephyrproject-rtos/zephyr/pull/78089) for some reason?
